### PR TITLE
Added parsing for PDB string in PE module

### PR DESCRIPTION
**Pull request by cherepanov74**
Implements this feature request: https://github.com/VirusTotal/yara/issues/736

Now it's possible to do following rules:
```
import "pe"

rule pdb_rule
{
   condition:
     pe.pdb contains "substring"
}
```

### CODE REVIEW

**Comment by cherepanov74** [comment]
I signed Google CLA

**Comment by nyx0** [comment]
@plusvic Do you have an ETA for merging? thank you :)

**Comment by plusvic** [comment]
> @plusvic Do you have an ETA for merging? thank you :)

Probably for next release, that is being baked right now. I have some comments about the latests changes in this PR though.

File: libyara/modules/pe.c [hunk 1]
```diff
@@ -317,7 +321,7 @@ static void pe_parse_debug_directory(
       if (!struct_fits_in_pe(pe, pdb20, CV_INFO_PDB20))
         break;
       
-      pdb_str_len = strlen((char *)(pdb20->PdbFileName));
+      pdb_str_len = strnlen((char *)(pdb20->PdbFileName), MAX_PATH * sizeof(char));
```
Comment (plusvic) [inline libyara/modules/pe.c]: This is is an improvement over using `strlen` but it doesn't solve the original issue as you still can read out of the buffer's bounds. You are limiting the number of bytes being read to MAX_PATH, but the bytes remaining in the buffer can still be less than that. So, the argument you you pass to strnlen is the space available from the pdb20->PdbFileName pointer to the end of the buffer. You can use the `available_space` function for computing that.

File: libyara/modules/pe.c [hunk 2]
```diff
@@ -261,6 +261,95 @@ static void pe_parse_rich_signature(
...
+      }
+    }
+    else if (yr_le32toh(cv_hdr->dwSignature) == CVINFO_PDB70_CVSIGNATURE)
+    {
+      PCV_INFO_PDB70 pdb70 = (PCV_INFO_PDB70)cv_hdr;
+      
+      if (!struct_fits_in_pe(pe, pdb70, CV_INFO_PDB70))
+        break;
+      
+      pdb_str_len = strnlen((char *)(pdb70->PdbFileName), MAX_PATH * sizeof(char));
```
Comment (plusvic) [inline libyara/modules/pe.c]: Same here.

**Comment by cherepanov74** [comment]
Thank you for all your suggestions!

**Pull request merged**